src/middleware/checkPermissionMiddleware.ts

```diff
@@ -0,0 +1,19 @@
+import { Request, Response, NextFunction } from 'express';
+
+interface User {
+  role: string;
+}
+
+interface CustomRequest extends Request {
+  user?: User;
+}
+
+const checkPermission =
+  (role: string) => (req: CustomRequest, res: Response, next: NextFunction) => {
```

Comment:
Why have this variable `role` if you are not using it?

Reply:
I am using it. 
I have made this function reusable so that everyone who want to use this middleware , must call it passing required role.

here is how I am using it  in [productRoute.ts](src/routes/productRoutes.ts)

`>
router.post('/', isAuthenticated, checkPermission('SELLER'), createCollection);
router.post(
  '/:collectionId',
  isAuthenticated,
  checkPermission('SELLER'),
  upload.array('images'),
  createProduct,`

<img width="518" alt="image" src="https://github.com/atlp-rwanda/e-commerce-ones-and-zeroes-bn/assets/97307891/4cd215e4-6a7e-4aa6-82f5-0d1816fda409">